### PR TITLE
Add RETURNING to INSERT and DELETE

### LINKED ISSUE #516: Document the `RETURNING` clause in DML statements
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/7094

### Which part(s) of the docs might be affected or should be updated? And how?

Read the code PR to evaluate. My initial evaluation is that DELETE and INSERT need to be updated with the clause. 

### Reference

https://www.postgresql.org/docs/current/dml-returning.html
https://lornajane.net/posts/2016/the-returning-keyword-in-postgresql

### PR DESCRIPTION
**Pull request by bernscode**
Add RETURNING to INSERT and DELETE

## Info
- **Description**: 
Add RETURNING to INSERT and DELETE.

- **Preview**: 
https://pr-540.d2fbku9n2b6wde.amplifyapp.com/docs/upcoming/sql-insert/
https://pr-540.d2fbku9n2b6wde.amplifyapp.com/docs/upcoming/sql-delete/

- **Related code PR**: 
https://github.com/risingwavelabs/risingwave/pull/7094

- **Related doc issue**: 
Resolves https://github.com/risingwavelabs/risingwave-docs/issues/516

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

### CODE REVIEW

**Comment by y-wei** [comment]
LGTM, thanks!

**Review by y-wei — APPROVED** [review]

**Review by y-wei — CHANGES_REQUESTED** [review]
Apologies for the unclear information in my PR. Actually ```UPDATE``` should also be updated along with ```INSERT``` and ```DELETE``` (add ```RETURNING```).

File: docs/sql/commands/sql-delete.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ Use the `DELETE` command to permanently remove rows from a table.
 
 ```sql
 DELETE FROM table_name
-WHERE condition;
+WHERE condition [ RETURNING col_name ];
```
Comment (CharlieSYH) [inline docs/sql/commands/sql-delete.md]: In the syntax, please use a new line for RETURNING.
See this example: https://www.postgresql.org/docs/current/dml-returning.html

File: docs/sql/commands/sql-insert.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ For tables with primary keys, if you insert a row with an existing key, the new
 
 ```sql
 INSERT INTO table_name [ ( col_name [ , ... ] ) ]
-      { VALUES ( value [ , ... ] ) [ , ( ... ) ] | select_query };
+      { VALUES ( value [ , ... ] ) [ , ( ... ) ] | select_query } [ RETURNING col_name ];
```
Comment (CharlieSYH) [inline docs/sql/commands/sql-insert.md]: Use a new line for RETURNING.

**Review by CharlieSYH — APPROVED** [review]
Approved with two comments.

File: docs/sql/commands/sql-delete.md (hunk 1 above)
Comment (bernscode) [reply docs/sql/commands/sql-delete.md]: @CharlieSYH Good point! I've made the update to DELETE AND INSERT. Thanks!

**Review by y-wei — APPROVED** [review]
LGTM

**Pull request merged**